### PR TITLE
add Minimizer plugin and System.minimize method

### PR DESCRIPTION
**Pull request by njzjz**
This PR adds the `minimize` method to `System` and also the Minimizer plugins. Also add two minimizers: ASE and AMBER SQM.

### CODE REVIEW

File: dpdata/driver.py [hunk 1]
```diff
@@ -81,6 +83,35 @@ def label(self, data: dict) -> dict:
...
+        system = dpdata.System(data=data)
+        # list[Atoms]
+        structures = system.to_ase_structure()
+        labeled_system = dpdata.LabeledSystem()
+        for atoms in structures:
+            atoms.calc = self.ase_calculator
+            dyn = LBFGS(atoms, logfile=None)
+            dyn.run(fmax=5e-3)
+            ls = dpdata.LabeledSystem(atoms, fmt="ase/structure", type_map=data['atom_names'])
+            labeled_system.append(ls)
```
Comment (wanghan-iapcm) [inline dpdata/driver.py:112]: Does one have any chance of specifying the optimizer and the parameters for optimization?

Comment (njzjz) [reply dpdata/driver.py:112]: Good point, let me consider how to design the API

Comment (njzjz) [reply dpdata/driver.py]: The API is refactored.

**Review by wanghan-iapcm — COMMENTED** [review]
Could you please provide doc on how to use the minimizer?  
especially when one wants to change the default `fmax` of "ase".

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**